n3fit/src/n3fit/layers/preprocessing.py

```diff
@@ -75,20 +72,25 @@ def generate_weight(self, name: str, kind: str, dictionary: dict, set_to_zero: b
         """
         constraint = None
         if set_to_zero:
-            initializer = MetaLayer.init_constant(0.0)
+            single_replica_initializer = MetaLayer.init_constant(0.0)
             trainable = False
         else:
             minval, maxval = dictionary[kind]
             trainable = dictionary.get("trainable", True)
-            # Set the initializer and move the seed one up
-            initializer = MetaLayer.select_initializer(
-                "random_uniform", minval=minval, maxval=maxval, seed=self.seed
+            # Seeds will be set in the wrapper MultiInitializer
+            # Note: keras 3 interprets a seed of 0 as None and replaces it with a random seed,
+            # so we set it to 1 here and subtract it later when the replica seed is set
```

Comment:
where is it set to 1?

Reply:
Sorry this was an earlier approach I tried, will remove this.